Board and UnitBoard layout changes

brianries opened the pull request:
The scope of this project was to make scaling changes to the UnitBoard and the Border Edges around the board.  The goal was to enforce some more strict tiling / sizing of the game cards given the various resolutions.  We want to prevent units crossing the center line *for almost all cases* but it will still cross the center on small resolution and very large unit counts (8+).  I also created a debug class - BreakPointOverlay so you can enable it and get visual feedback on which 'responsive threshold' you're operating with

A few additional things had to be fixed after this rescaling:
- I added a 3rd segment to the border outlines that can dynamically scale (it's just a horizontal line that fills inbetween the left and right side).  I scaled down the pixel width of the small curve, and let the horizontal line fill the rest.
- I fixed the upgrade icon / text on smaller images.  You can see in the first image that it was rendering more box then it needed.  This is still not perfect and should probably get redesigned, but it works.

Here is a good before and after, using the iPadAir resolution:

iPadAir before:
![image](https://github.com/user-attachments/assets/e1914758-f404-48fa-a893-c1cc2f4799ad)

iPadAir after:
![image](https://github.com/user-attachments/assets/ddf50b30-8a71-4d02-b676-eb00d0e5766a)

iPhone12 before:
![image](https://github.com/user-attachments/assets/947af449-6ff4-440a-bdf0-0bb8009d0282)

iPhone12 after:  (Still needs work, but closer!)
![image](https://github.com/user-attachments/assets/bb8ad19a-ee59-4732-84b5-f4545578673a)

Example of the border separating on wider monitors (this is now fixed by the 3rd segment that fills the gap)
![image](https://github.com/user-attachments/assets/a5cc51a4-42c5-4f89-ab91-431d463a3ad6)

danbastin commented:
I like this as a start. One think to look at is how damaged and "mid-card" tokens like sentinel get rendered. 

![image](https://github.com/user-attachments/assets/f91c81f7-ffcd-4d3b-8983-e15c1a2553e4)

This is using devtools iPad Air resolution. I think for something like this we could go a little bigger with the card and a little smaller with the tokens. 

Probably something we want to look into as well is stacking token units on smaller screens. Things like a crazy Endless Legion are always going to be tough but I think thats enough of an edgecase that would could have something like scrolling for the unit containers that is a less graceful handling but fine due to the rarity of the situation. Token units are a more likely situation so stacking toekns that are all the same stats could be an option.

AMMayberry1 commented:
Looks like the "default" size for everything got smaller, even on a mostly-empty board and a PC screen (see comparison below). I might just be old and going blind but I liked things being a bit bigger when possible - would it be doable to get back to that without messing up the scaling on smaller screens? Also good to gauge whether people other than me would like that, I guess.

![image](https://github.com/user-attachments/assets/532e640f-df85-4749-9c10-71148062c229)

![image](https://github.com/user-attachments/assets/ed69bb79-352b-4ebb-ba30-675f9c7fdf17)

brianries commented:
> I like this as a start. One think to look at is how damaged and "mid-card" tokens like sentinel get rendered.
> 
> ![image](https://private-user-images.githubusercontent.com/5757619/433548795-f91c81f7-ffcd-4d3b-8983-e15c1a2553e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ3NjkzMjYsIm5iZiI6MTc0NDc2OTAyNiwicGF0aCI6Ii81NzU3NjE5LzQzMzU0ODc5NS1mOTFjODFmNy1mZmNkLTRkM2ItODk4My1lMTVjMWEyNTUzZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTZUMDIwMzQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTE5NjM3ZjE4YTYxOWUxYzU4ZWIwMmIwMzk1M2NhZmM2MWVmYzA0MTJlY2RhNGY1ZWJiMTA2OThlOTNhMDZiZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FM9nzPRmDDJCOJxbCdsBkRWhFRv1lisQYBvxKjnB3NM)
> 
> This is using devtools iPad Air resolution. I think for something like this we could go a little bigger with the card and a little smaller with the tokens.
> 
> Probably something we want to look into as well is stacking token units on smaller screens. Things like a crazy Endless Legion are always going to be tough but I think thats enough of an edgecase that would could have something like scrolling for the unit containers that is a less graceful handling but fine due to the rarity of the situation. Token units are a more likely situation so stacking toekns that are all the same stats could be an option.

Updated the sizing of the tokens and converted to "em" and percentage space.  Upped the size of the cards a tiny bit at the ipadAir resolution level.

brianries commented:
> Looks like the "default" size for everything got smaller, even on a mostly-empty board and a PC screen (see comparison below). I might just be old and going blind but I liked things being a bit bigger when possible - would it be doable to get back to that without messing up the scaling on smaller screens? Also good to gauge whether people other than me would like that, I guess.
> 
 
 Upped the size of the game cards to 7rem for this setting (don't shrink them just yet)

AMMayberry1 commented:
Guessing you didn't mean to leave this enabled (though it was very helpful for the test):

![image](https://github.com/user-attachments/assets/341d6c73-6d62-4cbd-a13a-0222f542eaaf)
